src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -149,6 +156,16 @@ We support
             }
             else
             {
+                string rawRequirementType = configurationSection[RequirementTypeKeyword];
+
+                if (!string.IsNullOrEmpty(rawRequirementType))
+                {
+                    if (!Enum.TryParse(rawRequirementType, true, out requirementType))
+                    {
+                        throw new ArgumentException($"Invalid requirement type '{rawRequirementType}' for feature '{configurationSection.Key}'.");
```

Comment:
Technically, rawRequirementType is not an argument so we shouldn't throw ArgumentException.
I think it should be okay to add a new error type for `InvalidConfiguration` in FeatureManagementException.